guides/svg/template.md

```diff
@@ -0,0 +1,229 @@
+
+**Important :** you need to set the `viewBox` attribute to be equal to your icon width and height.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+  
+    <svg id="icon-{{id}}"
+         class="component-icon"
+         width="32" height="32"></svg>
+
+    <text class="component-name" y="1em">{{ name }}</text>
+    <text class="component-type" y="2.5em">{{ definition.type }}</text>
+
+    <svg class="component-container" overflow="visible">
+        <rect class="container-background" width="100%" height="100%" fill="whitesmoke"/>
+    </svg>
+
+    <!-- Error icon-->
+    {% if hasError %}
+    <g class="component-error">
+        <!-- This is a placeholder, you can define the icon you want by replacing this svg. -->
+        <svg width="15" height="15" viewBox="0 0 100 100">
+            <circle fill="red" r="50" cx="50" cy="50"></circle>
+        </svg>
+    </g>
+    {% endif %}
+</svg>
+```
+
+### 7. Position the elements :
+Now that we have all the elements, we can position them.
+We can't use the `x` and `y` attributes, because the FireFox browser have weird behavior with them. So we use the
+`<g></g>` element and the `transform` attribute with the `translate(x,y)` function, to have a new position reference
+frame for each element of our component.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+    
+    <!-- Position the Icons and Metadata -->
+    <g tranform="translate(5,5)">
+        <svg id="icon-{{id}}" class="component-icon" width="32" height="32"></svg>
+        <g transform="translate(32,0)">
+            <text class="component-name" y="1em">{{ name }}</text>
+            <text class="component-type" y="2.5em">{{ definition.type }}</text>
+        </g>
+    </g>
+    
+    <!-- Position the container -->
+    <g transform="transform(0,50)">
+        <svg class="component-container" overflow="visible">
+            <rect class="container-background" width="100%" height="100%" fill="whitesmoke"/>
+        </svg>
+    </g>
+
+    {% if hasError %}
+    <!-- Position the error icon -->
+    <g class="component-error" transform="translate(150,5)">
+        <svg width="15" height="15" viewBox="0 0 100 100">
+            <circle fill="red" r="50" cx="50" cy="50"></circle>
+        </svg>
+    </g>
+    {% endif %}
+</svg>
+```
+
+---
+
+## Tips and Tricks
+
+### Center an element from component width and height :
+To center an element, you need to use the `transform` attribute with the `translate(x,y)` function, and the `x` and `y`
+of the element.
+
+```jsx
+<svg class="template" width="200" height="200">
+    <!-- Center the rectangle :
+    - Set x and/or y to 50% to center from width and/or height
+    - Set overflow to visible to be able to see the SVG when it's outside the svg
+    -->
+    <svg x="50%" y="50%" overflow="visible">
```

Comment:
is that working on firefox ?

Reply:
It works, but I forgot to specify that you have to set a size on the `svg` that contains the element to center, but I corrected it.